### PR TITLE
Reorganize the CD ripper

### PR DESCRIPTION
**Pull request by paperbagcorner**
The code for the CD ripper has become somewhat unwieldy. I have therefore split the ripper into two classes, one RipCDDialog class for the user interface and one Ripper class that handles the ripping.

The Rip audio CD menu item is moved to the Tools menu since I believe it is a more natural place.

The commit f7185101cea9d672a2fc4afc16d389e8afbd832f fixes a bug were the wrong file was tagged after the transcoder had changed the output filename.

### CODE REVIEW

File: src/ripper/ripcddialog.cpp [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+  setWindowTitle(tr("Rip CD"));
+  AddDestinationDirectory(QDir::homePath());
+
+  // Get presets
+  QList<TranscoderPreset> presets = Transcoder::GetAllPresets();
+  qSort(presets.begin(), presets.end(), ComparePresetsByName);
+  for (const TranscoderPreset& preset : presets) {
+    ui_->format->addItem(
+        QString("%1 (.%2)").arg(preset.name_, preset.extension_),
```
Comment (hatstand) [inline src/ripper/ripcddialog.cpp]: `.arg(a).arg(b)` is preferred as it avoids future bugs when formatting numbers due to the multiple overloads of `arg()`

File: src/ripper/ripcddialog.cpp [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+}
+
+void RipCDDialog::InvertSelection() {
+  for (QCheckBox* checkbox : checkboxes_) {
+    if (checkbox->isChecked()) {
+      checkbox->setCheckState(Qt::Unchecked);
+    } else {
+      checkbox->setCheckState(Qt::Checked);
+    }
+  }
```
Comment (hatstand) [inline src/ripper/ripcddialog.cpp]: `checkbox->setCheckState(checkbox->isChecked() ? Qt::Unchecked : Qt:Checked);`

File: src/ripper/ripper.cpp [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+    cancel_requested_ = true;
+  }
+  transcoder_->Cancel();
+  RemoveTemporaryDirectory();
+  emit(Cancelled());
+}
+
+void Ripper::TranscodingJobComplete(const QString& input, const QString& output,
+                                    bool success) {
+  (*(success ? &finished_success_ : &finished_failed_))++;
```
Comment (hatstand) [inline src/ripper/ripper.cpp]: I think a simple `if` statement is more readable.

File: src/ripper/ripper.cpp [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+  UpdateProgress();
+
+  // The the transcoder does not overwrite files. Instead, it changes
+  // the name of the output file. We need to update the transcoded
+  // filename for the corresponding track so that we tag the correct
+  // file later on.
+  for (TrackInformation& track : tracks_) {
```
Comment (hatstand) [inline src/ripper/ripper.cpp]: Don't use non-const references

File: src/ripper/ripper.cpp (hunk 3 above)
Comment (paperbagcorner) [inline src/ripper/ripper.cpp]: It is and you've commented on in [before](https://github.com/clementine-player/Clementine/pull/4113/files#r8858034). I will change it.

Comment (hatstand) [inline src/ripper/ripper.cpp]: I thought I recognised it :-)

**Pull request merged**